Tests/Common/Helpers/FakeURLProtocol.swift

```diff
@@ -0,0 +1,82 @@
+import Foundation
+
+/// A fake implementation of URLProtocol.
+/// This class is heavily adapted from https://jeroenscode.com/mocking-alamofire/.
+final class FakeURLProtocol: URLProtocol {
+  // Types of responses.
+  enum ResponseType {
+    // An error response and associated value.
+    case error(Error)
+
+    // A success response type, with the given response and ASCII encoded data.
+    case success(HTTPURLResponse, String)
+  }
+  static var responseType: ResponseType!
+
+  private lazy var session: URLSession = {
+    let configuration: URLSessionConfiguration = URLSessionConfiguration.ephemeral
+    return URLSession(configuration: configuration, delegate: self, delegateQueue: nil)
+  }()
+  private(set) var activeTask: URLSessionTask?
+
+  override class func canInit(with request: URLRequest) -> Bool {
```

Comment:
Some comments on these methods would be helpful to understand context on what these are for and how they are used together.

Reply:
I added docs on the stuff that is unique. 

A lot of this is boilerplate implementations of protocols, which already have the doc. For instance, there's official documentation on [URLProtocol](https://developer.apple.com/documentation/foundation/urlprotocol) and the [`canInit(with:)` method](https://developer.apple.com/documentation/foundation/urlprotocol/1411389-caninit).

I attached `SeeAlso` tags to link to this documentation. 

LMK if you want more detail. I'm being somewhat vague because I don't really know the way all of these methods wires together and I'm really just following the linked tutorial.